fuzz/fuzz_targets/parse_hrp.rs

```diff
@@ -0,0 +1,59 @@
+extern crate bech32;
+
+use bech32::Hrp;
+
+fn do_test(data: &[u8]) {
+    let s = String::from_utf8_lossy(data);
+
+    // Make sure parsing garbage doesn't make us crash (from_utf8_lossy should
+    // contain some garbage, perhaps even invalid chars).
+    let _ = Hrp::parse(&s);
```

Comment:
`from_utf8_lossy` will not include any invalid characters, at least not in the UTF8 sense.

I think we should use [str::from_utf8[(https://doc.rust-lang.org/std/str/fn.from_utf8.html) and return on error. This avoids an allocation and avoids mutating the input string, so it should be faster.

Reply:
oooo, it replaces invalid chars with `?` - that's where those all came from when I tried to iterate over all the zeros in the buffer :)

> Converts a slice of bytes to a string, including invalid characters.

ref: https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_lossy

I think your right, I'll try with `from_utf`. Thanks